bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/ConnectionValidator.java

```diff
@@ -74,15 +62,16 @@ public static void validateConnections(RuntimeRule r) {
      * @param actions is a list with actions of the rule whose connections have to be validated
      * @throws IllegalArgumentException when validation fails.
      */
-    public static void validateConnections(List<Trigger> triggers, List<Condition> conditions, List<Action> actions) {
+    public static void validateConnections(ModuleTypeRegistry mtRegistry, List<TriggerImpl> triggers,
+            List<ConditionImpl> conditions, List<ActionImpl> actions) {
         if (!conditions.isEmpty()) {
-            for (Condition condition : conditions) {
-                validateConditionConnections((RuntimeCondition) condition, triggers);
+            for (ConditionImpl condition : conditions) {
+                validateConditionConnections(mtRegistry, condition, triggers);
             }
         }
         if (!actions.isEmpty()) {
             for (Action action : actions) {
-                validateActionConnections((RuntimeAction) action, triggers, actions);
+                validateActionConnections(mtRegistry, (ActionImpl) action, triggers, actions);
```

Comment:
Why not using `for (ActionImpl action : actions)` instead of using the interface type for the element taken from the `ActionImpl` list and then cast again?

Reply:
I agree, will update the pull request.